python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -856,6 +863,22 @@ def evaluate_tick(self, context: "SensorEvaluationContext") -> "SensorExecutionD
             run_requests, context, self._asset_selection, dynamic_partitions_requests
         )
 
+        for backfill_request in backfill_requests:
+            asset_selection = check.not_none(
+                self._asset_selection,
+                "Can only yield BackfillRequests for sensors with an asset_selection",
+            )
+            asset_keys = backfill_request.asset_keys
+
+            unexpected_asset_keys = (AssetSelection.keys(*asset_keys) - asset_selection).resolve(
+                check.not_none(context.repository_def).asset_graph
+            )
+            if unexpected_asset_keys:
+                raise DagsterInvalidSubsetError(
+                    "BackfillRequest includes asset keys that are not part of sensor's asset_selection:"
+                    f" {unexpected_asset_keys}"
+                )
```

Comment:
confirming understanding - this validation happens before we attempt to launch any runs or backfills (which is important so that if you have a sensor that launches a run and a backfill, or two backfills, and the first one succeeds but the second one finishes, we don't end up in a submit loop where we are launching the first thing)